### DISCUSSION TITLE
Best way to get a list of thumb drives connected to Raspberry Pi OS

### DISCUSSION THREAD

**Discussion by coobr01** [pyusb/pyusb#435] [Q&A] [answered]
Situation: Building a flask based web application in python and would like to get a list of USB thumb drives connected to the Raspberry Pi OS (I won't know the name, vendor, etc. for the drives beforehand). For each thumb drive found, I would like to obtain any name that has been assigned to them and their "system" address/path (e.g., /media/pi/B3C9-4CD4) so that I can use other Python packages (e.g., Pandas) to write files to one of the thumb drives.

**Comment by jonasmalacofilho** [comment, marked as answer]
The most comprehensive/flexible way to do this is probably by inspecting `sysfs`.  But a simpler way might be to start with grepping the output of lsblk for removable drives:[^1]

```
$ lsblk --pairs | grep 'RM="1"'
```

Either way, trying to do it with PyUSB would only make things unnecessarily complicated, as in your scenario you don't really care about how those devices work at the USB level.

[^1]: https://stackoverflow.com/a/70601544

**Comment by coobr01** [comment]
Thank you for the information!
